nixos/modules/services/web-apps/trilium.nix

```diff
@@ -0,0 +1,151 @@
+  '';
+in
+{
+
+  options.services.trilium-server = with lib; {
+    enable = mkEnableOption "trilium-server";
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/trilium";
+      example = "/var/lib/trilium";
+      description = ''
+        The directory storing the nodes database and the configuration.
+      '';
+    };
+
+    instanceName = mkOption {
+      type = types.str;
+      default = "Trilium";
+      example = "Trilium";
+      description = ''
+        Instance name used to distinguish between different instances
+      '';
+    };
+
+    host = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      example = "127.0.0.1";
+      description = ''
+        The host address to bind to (defaults to localhost).
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 8080;
+      example = 8080;
+      description = ''
+        The port number to bind to.
+      '';
+    };
+
+    nginx = mkOption {
+      default = {};
+      description = ''
+        Configuration for nginx reverse proxy.
+      '';
+
+      type = types.submodule {
+        options = {
+          enable = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              Configure the nginx reverse proxy settings.
+            '';
+          };
+
+          hostName = mkOption {
+            type = types.str;
+            default = "";
+            description = ''
+              The hostname use to setup the virtualhost configuration
+            '';
+          };
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable (lib.mkMerge [ 
+  { assertions = [
+      { assertion = cfg.nginx.enable == true -> cfg.nginx.hostName != "";
+          message = "Nginx host name cannot be empty when reverse proxy settings are enabled";
+      }
+    ];
+  }
+  {
+    meta.maintainers = with lib.maintainers; [ kampka ];
+
+    users.groups.trilium = {};
+    users.users.trilium = {
+      description = "Trilium User";
+      group = "trilium";
+      home = cfg.dataDir;
+      isSystemUser = true;
+    };
+
+    systemd.services = {
+      trilium-server = {
+          wantedBy = [ "multi-user.target" ];
+          script = ''
+            exec ${pkgs.trilium-server}/bin/trilium-server
+          '';
+          serviceConfig = {
+            Environment = [ "TRILIUM_DATA_DIR=${cfg.dataDir}" ];
+            User = "trilium";
+            Group = "trilium";
+            PrivateTmp = "true";
+          };
+      };
+
+    };
+
+    systemd.tmpfiles.rules = [
+      "d  ${cfg.dataDir}            0750 trilium trilium - -"
+      "L+ ${cfg.dataDir}/config.ini -    -       -       - ${configIni}"
+    ];
+
+  }
+
+  (lib.mkIf cfg.nginx.enable {
+    services.nginx = {
+      enable = true;
+      virtualHosts."${cfg.nginx.hostName}" = {
+        locations."/".extraConfig = ''
+          proxy_pass http://${cfg.host}:${toString cfg.port}/;
```

Comment:
Having never used `nginx` or the module: can you use `locations<name>.proxyPass` here in addition to `extraConfig`? Don't consider this a change request, but informational for me because I'm working on a PR were I need to better understand the `nginx` module.

Reply:
You can use both `proxyPass` and `extraConfig`, I didn't feel like there is a huge need to do this here as the extraConfig block is already quite huge by itself. Refactored it anyways.

Reply:
Thanks for explaining.